R/doeResponseSurfaceMethodology.R

```diff
@@ -787,26 +812,113 @@ doeResponseSurfaceMethodology <- function(jaspResults, dataset, options, ...){
 
 .responseSurfaceOptimize <- function(jaspResults, options, rsm, data, position, dataset) {
 
-  unique_l <- list()
-  for (i in 1:(ncol(data)-1)) {
-    unique_l[[i]] <- unique(data[,i])
+  desire_list <- list()
+  k <- 1
+  if (length(options[["rsmMin"]]) > 0) {
+    for (i in 1:length(options[["rsmMin"]])) {
+      desire_list[[k]] <- desirability::dMin(options[["rsmMin"]][[i]][[2]], options[["rsmMin"]][[i]][[1]])
+      k <- k +1
+    }
+  }
+  if (length(options[["rsmMax"]]) > 0) {
+    for (i in 1:length(options[["rsmMax"]])) {
+      desire_list[[k]] <- desirability::dMin(options[["rsmMax"]][[i]][[2]], options[["rsmMax"]][[i]][[1]])
+      k <- k +1
+    }
+  }
+  if (length(options[["rsmTar"]]) > 0) {
+    for (i in 1:length(options[["rsmTar"]])) {
+      desire_list[[k]] <- desirability::dMin(options[["rsmTar"]][[i]][[2]],options[["rsmTar"]][[i]][[3]], options[["rsmTar"]][[i]][[1]])
+      k <- k +1
+    }
+  }
+
+  desire_final <- do.call(desirability::dOverall, desire_list)
+
+  if (length(options[["rsmMax"]]) + length(options[["rsmMin"]]) + length(options[["rsmTar"]]) == 1)
+    desire_final <- desire_final[[1]][[1]]
+
+  op  <- data.frame(x1 = seq(-1,1,0.25))
+  op1 <- length(options[["rsmVariables"]])
+
+  if (op1 > 1) {
+    for (i in 2:op1) {
+      op <- cbind(op, seq(-1,1,0.25))
+    }
+    names(op) <- paste0("x",1:op1)
   }
-  # print(unique_l)
-  comb <- expand.grid(unique_l)
-  # print(names(data))
-  unique_l2 <- list()
-  for (i in 1:(ncol(data)-1)) {
-    unique_l2[[i]] <- unique(data[,i])
+
+
+  possibleCodedCombinationsOfPredictors <- expand.grid(op)
+
+  rsmOpt_2 <- function(predictorCombinationFunction, rsm, dObject, space = "square") {
+    conv <- numeric(length(rsm))
+    for (j in seq_along(rsm)) {
+      conv[j] <- predict(rsm[[j]], newdata = possibleCodedCombinationsOfPredictors[i,])
+    }
+    out <- do.call(predict, list(object = dObject, newobject = conv))
+
+    if (space == "circular") {
+      if (sqrt(sum(possibleCodedCombinationsOfPredictors[i,]^2)) > 1.682)
```

Comment:
where do these magic numbers (1.682) come from? maybe leave a comment explaining this?